hw3/hw3-react/src/views/Search.js

```diff
@@ -0,0 +1,75 @@
+import React, { useState, useEffect } from 'react';
+
+function Search(props) {
+  const [post, setPost] = useState('');
+  const [data, setData] = useState([]);
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [match, setMatch] = useState(true);
+  const [toggle, setToggle] = useState(false);
+  const { title } = props;
+
+  let [name, setName] = useState('');
+  let [pic, setPic] = useState('');
+
+  const url = 'https://thronesapi.com/api/v2/Characters';
+  useEffect(() => {
+    fetch(url)
+      .then((res) => res.json())
+      .then((characters) => {
+        setData(characters);
+        setIsLoaded(true);
+      })
+      .catch((error) => console.log(error));
+  }, []);
```

Comment:
Calls the same API and fetches the same data as that of Houses.js

Fetching the data (calling the exact same API) twice, in both the Houses and Search components is inefficient. The data they both rely on is the same, and additionally both of these components have the same parent (should be something like App.js, not Navbar).

So why not let the data flow from parent to child?

It would make a lot more sense to call the API once in the App component, then pass the data as props to the houses and search components. This would be more efficient since the API would only need to be called once, but the data would be available across the components that need it.

In summary, you can just fetch the data once and then pass it to child components via props.